src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,42 @@
+package seedu.addressbook.data.person;
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact {
+    public String value;
+    private boolean isPrivate;
+    
+    public Contact (String val, boolean isPriv, String constraints, String validation) throws IllegalValueException {
+    	this.isPrivate = isPriv;
+    	String trimmedContact = val.trim();
+    	if (!isValidContact (trimmedContact, validation)) {
+    		throw new IllegalValueException(constraints);
+    	}
+    	this.value = trimmedContact;
+    }
+    
+    public boolean isValidContact (String test, String validation) {
+    	return test.matches(validation);
+    }
+    
+    @Override
+    public String toString() {
+    	return value;
+    }
+    
+    @Override
+    public boolean equals (Object other) {
+    	return other == this 
+    			|| (other instanceof Contact
+    			&& this.value.equals(((Contact)other).value));	
```

Comment:
Note that now an Email and Address having the same value will be considered equal even if they are of different types.